Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/CwaCovidCertificate.kt

```diff
@@ -105,6 +107,17 @@ interface CwaCovidCertificate : Recyclable {
                 .registerSubtype(ExpiringSoon::class.java, "ExpiringSoon")
                 .registerSubtype(Expired::class.java, "Expired")
                 .registerSubtype(Invalid::class.java, "Invalid")
+                .registerSubtype(Blocked::class.java, "Blocked")
+        }
+
+        override fun equals(other: Any?): Boolean {
+            if (this is Blocked && other is Blocked) return true
+            if (this is Recycled && other is Recycled) return true
```

Comment:
why do you need special handling for these cases ?

Reply:
comparing state with deserialized state didn't work as expected in https://github.com/corona-warn-app/cwa-app-android/blob/747cf79192f218129e4d9fbdad8de8fb8d8ab9ca/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/vaccination/core/repository/storage/VaccinationContainer.kt#L175

even if `state` and `lastSeenStateChange` were objects of the same type `Blocked`, equals returned false. It works for data classes.